pycontrails/core/models.py

```diff
@@ -731,25 +762,271 @@ def interpolate_met(
     KeyError
         Parameter ``met_key`` not found in ``met``.
     """
-    if vector_key is None:
-        vector_key = met_key
+    vector_key = vector_key or met_key
 
     if (out := vector.get(vector_key, None)) is not None:
         return out
 
     if met is None:
-        raise ValueError(f"No variable key {vector_key} in ``vector`` and ``met`` is None")
+        raise ValueError(f"No variable key '{vector_key}' in 'vector' and 'met' is None")
+
+    if met_key in ("q", "specific_humidity"):
+        mda, log_applied = _extract_q(met, met_key, q_method)
```

Comment:
this feels like it should be within `interpolate_gridded_specific_humidity` and then you can get rid of `log_applied` by clearly separating the approaches into different sub-functions

Reply:
I've thought about this ... but it's not as easy as it sounds. We're calling `interpolate_gridded_specific_humidity` in a two different places, and the `log_applied` flag is determine upstream of this in one case. I agree that this is a little ugly as written, but I'm going to keep as is for now.